lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -255,7 +260,7 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
                 }
 
                 auto dstShaderInfo = _GetExportedShaderForNode(
-                    dstPlug.node(), materialExportPath, context, shaderWriterMap);
+                    dstPlug.node(), materialExportPath, context, shaderWriterMap, false);
```

Comment:
As far as I can see, the problem here is that we take destination plugs that we shouldn't. And it seems to me that it is at line 231 (if (!iterPlug.destinations(dstPlugs, &status) || status != MS::kSuccess) {) that we do that. So maybe the good solution would be to remove this code from there.

Reply:
Yes, that would be a more correct fix. We need to process only the connections returned by MItDependencyGraph and should not be looking for the full list of destination plugs. I lacked time to implement this complete solution and resorted to the quickest fix that produced correct results. If you have time to rewrite the function then it would improve the codebase.

Make sure to add unit test for both the issue in EMSUSD-549 and the other one I put in a comment above.

Reply:
@vlasovi  @JGamache-autodesk  Thanks for the feedback. I'll update the code and will add test cases to cover the above more scenarios.

I should mention even though this correctly exports as USD surface preview, it still doesn't have the support needed to make the ColorGain link survive. For this reason, the imported usd still does not fully reflect the exported ones.

Furthermore, if we switch to MaterialX when exporting, the original issue of EMSUSD-549 doesn't occur at all and the graph is exported and imported and the user's graph is preserved.

For that reason, the tests to compare export and import will automatically fail if added and we would have to skip that for the time being until we add support for fanned out nodes in USD Surface Preview scenarios, if at all possible.